### ISSUE TITLE
Events API endpoint not streaming complete JSON chunks.

### ISSUE THREAD

**Issue by joshmanders** [docker/docker#25831]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 5
 Running: 4
 Paused: 0
 Stopped: 1
Images: 27
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 66
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay host bridge null
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.954 GiB
Name: moby
ID: SSMC:OQPL:WUWW:2LRE:ZDLV:YHY3:4LMT:QTRI:KVB3:4JKN:USBQ:PV6E
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 32
 Goroutines: 53
 System Time: 2016-08-18T04:04:03.211239393Z
 EventsListeners: 2
No Proxy: *.local, 169.254/16
Username: joshmanders
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Describe the results you received:**

When doing a request to get the stream of events, sometimes a chunk is half a JSON string, and sometimes it's 3 strings combined.

When working on a events stream library for Docker in Node.js this makes it hard to parse the JSON string into an object that can be used to do things based on the event.

**Describe the results you expected:**

Each event streamed is it's own fully parsable JSON string.

**Comment by cpuguy83** [comment]
Can you provide an example?
Events should be a json stream, so basically a stream of event structs rather than a full/proper json object.

**Comment by joshmanders** [comment]
Yes, here is the code I am using, it worked last time I was working on this library, and I went to work on it to start using the events stream for monitoring and it doesn't work anymore, I don't know if it's the `request` library or the new docker API that changed. I can't figure out how to validate that the API is actually sending full JSON blobs for each chunk.

https://github.com/aniftyco/docker/blob/master/src/docker.js#L69-L86

**Comment by cpuguy83** [comment]
Not really familiar with js API's here...
But you'll need to use a streaming decoder that can actually understand the start/end of a streamed object.

**Comment by joshmanders** [comment]
Do you have any suggestions? I'd really like to pinpoint this issue and correct it. Spent 6 hours googling and trying to fix this before making this issue. Was my last ditch effort. :(

**Comment by cpuguy83** [comment]
Seems like there's a bunch of javascript libraries for docker. 

https://www.npmjs.com/package/docker-events
https://www.npmjs.com/package/docker-event-stream
https://www.npmjs.com/package/dockerode

**Comment by joshmanders** [comment]
Not sure what you're saying regarding that... dockerode is the only one that has been updated in the last year and has tests, which are sparse. I am going in a different direction.

**Comment by cpuguy83** [comment]
@joshmanders I can't tell you how to write your javascript (been ages since I've even touched js), what I can tell you is that relying on the events response and assuming the response is a full object is probably not right.
`/events` is a streaming API as such you can't just unmarshal the response, you need to decode the json stream.

**Comment by joshmanders** [comment]
Thank you for your assistance @cpuguy83, I was wrong it wasn't Docker.
